src/aims_ui/page_multiple_address.py

```diff
@@ -64,36 +78,36 @@ def final(searchable_fields,
     )
 
     file = request.files['file']
-    file_size = int(os.fstat(file.fileno()).st_size) / 1000000  # In MB
-    max_file_size = 1  # In MB
-
-    if file.filename == '':
-      return final(searchable_fields,
-                   error_description='Select a file that is a CSV ',
-                   error_title='File Type Error')
-
-    if file_size > max_file_size:
-      return final(
-          searchable_fields,
-          error_description=
-          f'File size is too large. Please enter a file no larger than {max_file_size} MB',
-          error_title='File Size Error')
 
-    if file and allowed_file(file.filename):
-      filename = secure_filename(file.filename)
+    file_valid, error_description, error_title = check_valid_upload(file)
 
+    if file_valid:
       for field in searchable_fields:
         if field.database_name == 'display-type':
           results_type = field.get_selected_radio()
 
       if results_type == 'Download':
         full_results, line_count = multiple_address_match(file, {},
                                                           download=True)
+        if full_results == 'error_connecting':
+          return page_error(None, page_name, connection_error=True)
 
         return send_file(full_results,
                          mimetype='text/csv',
                          attachment_filename=f'result_size_{line_count}.csv',
                          as_attachment=True)
+
       elif results_type == 'Display':
-        table_results = multiple_address_match(file, {}, download=False)
-        return final(searchable_fields, table_results=table_results)
+        table_results, results_summary_table = multiple_address_match(
+            file, {}, download=False)
+
+        if table_results == 'error_connecting':
```

Comment:
Likewise, this looks like more error-checking code that could just be a `catch`.